v4/key_test.go

```diff
@@ -18,6 +19,45 @@ import (
 	"github.com/miekg/pkcs11"
 )
 
+func init() {
+	var err error
+	ecPubKey, err = genECPubKey()
+	if err != nil {
+		log.Fatal(err)
+	}
+	ecPoint, err = getECPoint(ecPubKey.(*ecdsa.PublicKey))
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// Generate an ephemeral ECDSA key pair and return the public key corresponding
+// to the private key. If a key pair cannot be generated, an error is returned.
+func genECPubKey() (crypto.PublicKey, error) {
+	privateKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return nil, err
+	}
+
+	return privateKey.Public(), nil
+}
+
+// Retrieve the DER-encoding of ANSI X9.62 ECPoint value 'Q' from a given ECDSA
+// public key to match against a PKCS#11 CKA_EC_POINT attribute. If the
+// point cannot be marshalled, an error will be returned.
```

Comment:
Just running down the citations on this:

http://docs.oasis-open.org/pkcs11/pkcs11-curr/v2.40/cos01/pkcs11-curr-v2.40-cos01.html#_Toc408226932

> CKA_EC_POINT1,4 | Byte array | DER-encoding of ANSI X9.62 ECPoint value Q

X9.62 is paywalled but https://www.secg.org/sec1-v2.pdf appears to reproduce a lot of what's needed. In particular section 2.3.3 seems to contain encoding instructions.

Reply:
I think I was able to sufficiently address this with my most recent commit.